zcash_extras/src/wallet.rs

```diff
@@ -209,6 +247,11 @@ where
         }
     };
 
+    // Automatically decrypt and store any outputs sent to our wallet, including change.
+    // This uses our viewing keys to find any outputs we can decrypt, creates decrypted
+    // note data for spendability, and saves them to the wallet database.
+    decrypt_and_store_transaction(params, wallet_db, &tx).await?;
```

Comment:
Without this code we would need to wait until the block is scanned, right?

Reply:
right

Reply:
with the case above, we need this fix in kdf directly after building txs..